les/fetcher.go

```diff
@@ -205,8 +214,11 @@ func (f *lightFetcher) syncLoop() {
 		case p := <-f.syncDone:
 			f.lock.Lock()
 			p.Log().Debug("Done synchronising with peer")
-			f.checkSyncedHeaders(p)
+			res, h, td := f.checkSyncedHeaders(p)
 			f.syncing = false
+			if res {
```

Comment:
Is there a reason for factoring out the newHeaders call from checkSyncedHeaders?

Reply:
Yes, i fixed datarace ```WARNING: DATA RACE
Read at 0x00c4207da0b8 by goroutine 224:
  github.com/ethereum/go-ethereum/les.(*lightFetcher).checkUpdateStats()
      /Users/boris/go/src/github.com/ethereum/go-ethereum/les/fetcher.go:948 +0x277
  github.com/ethereum/go-ethereum/les.(*lightFetcher).unregisterPeer()
      /Users/boris/go/src/github.com/ethereum/go-ethereum/les/fetcher.go:248 +0xae
  github.com/ethereum/go-ethereum/les.(*peerSet).Unregister()
      /Users/boris/go/src/github.com/ethereum/go-ethereum/les/peer.go:601 +0x246
  github.com/ethereum/go-ethereum/les.testOdr()
      /Users/boris/go/src/github.com/ethereum/go-ethereum/les/odr_test.go:206 +0x8e2
  github.com/ethereum/go-ethereum/les.TestOdrGetBlockLes1()
      /Users/boris/go/src/github.com/ethereum/go-ethereum/les/odr_test.go:42 +0x56
  testing.tRunner()
      /usr/local/go/src/testing/testing.go:777 +0x16d

Previous write at 0x00c4207da0b8 by goroutine 102:
  github.com/ethereum/go-ethereum/les.(*lightFetcher).checkAnnouncedHeaders()
      /Users/boris/go/src/github.com/ethereum/go-ethereum/les/fetcher.go:703 +0x48a
  github.com/ethereum/go-ethereum/les.(*lightFetcher).newHeaders()
      /Users/boris/go/src/github.com/ethereum/go-ethereum/les/fetcher.go:627 +0x16c
  github.com/ethereum/go-ethereum/les.(*lightFetcher).syncLoop()
      /Users/boris/go/src/github.com/ethereum/go-ethereum/les/fetcher.go:221 +0x4ff
```